### ISSUE TITLE
[Bug] Flyout menu broken on latest version

### ISSUE THREAD

**Issue by operationslazlo326** [xamarin/Xamarin.Forms#7656]
When we use a combination of tabs with flyout the flyout dissappears on the latest build 4.2.0.815419 we are stuck on version 4.2.0.709249 until this is fixed.

See app shell XAML below.

<?xml version="1.0" encoding="UTF-8"?>
<Shell xmlns="http://xamarin.com/schemas/2014/forms" 
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
       xmlns:d="http://xamarin.com/schemas/2014/forms/design"
       xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
       mc:Ignorable="d"
       xmlns:xyz="clr-namespace:Xamarin.Forms;assembly=Xamarin.Forms.Core"
       xmlns:local="clr-namespace:com.enmarket.loyalty.consumer.Views"
       Title="com.enmarket.loyalty.consumer"
       xmlns:ext="clr-namespace:com.enmarket.loyalty.consumer"
       x:Class="com.enmarket.loyalty.consumer.AppShell"
       FlyoutBehavior="Flyout">

    
    <Shell.Resources>
        <ResourceDictionary>
            <xyz:Color x:Key="NavigationPrimary">#70BB44</xyz:Color>
            <Style x:Key="BaseStyle" TargetType="Element">
                <Setter Property="Shell.BackgroundColor" Value="{StaticResource NavigationPrimary}" />
                <Setter Property="Shell.ForegroundColor" Value="White" />
                <Setter Property="Shell.TitleColor" Value="White" />
                <Setter Property="Shell.DisabledColor" Value="#B4FFFFFF" />
                <Setter Property="Shell.UnselectedColor" Value="#95FFFFFF" />
                <Setter Property="Shell.TabBarBackgroundColor" Value="{StaticResource NavigationPrimary}" />
                <Setter Property="Shell.TabBarForegroundColor" Value="White"/>
                <Setter Property="Shell.TabBarUnselectedColor" Value="#95FFFFFF"/>
                <Setter Property="Shell.TabBarTitleColor" Value="White"/>
               
            </Style>
            <Style TargetType="ShellItem" BasedOn="{StaticResource BaseStyle}" />
            <Style TargetType="TabBar" BasedOn="{StaticResource BaseStyle}" />
          
        </ResourceDictionary>
    </Shell.Resources>

    <Shell.FlyoutHeaderTemplate>
        <DataTemplate>
            <StackLayout Orientation="Vertical" VerticalOptions="Start" HorizontalOptions="Start" BackgroundColor="White">
                <Image Source="{ext:Extensions.ImageResource com.enmarket.loyalty.consumer.Assets.Images.enlogo1.png}" />
            </StackLayout>
        </DataTemplate>
    </Shell.FlyoutHeaderTemplate>
    
    
    
    <TabBar>
        <Tab Title="HOME" Icon="tab_feed.png">
            <ShellContent ContentTemplate="{DataTemplate local:DealsPage}" />
        </Tab>
        <Tab Title="LOTTO" Icon="tab_feed.png">
            <ShellContent ContentTemplate="{DataTemplate local:LotteryPage}" />
        </Tab>
        <Tab Title="GIFTCARD" Icon="{ext:Extensions.ImageResource com.enmarket.loyalty.consumer.Assets.Images.giftcard-icon.png}" StyleClass="TabText">
            <ShellContent ContentTemplate="{DataTemplate local:GiftCardPage}" />
        </Tab>
        <Tab Title="REWARDS" Icon="tab_feed.png">
            <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
        </Tab>
        <Tab Title="SCAN" Icon="tab_feed.png">
            <ShellContent ContentTemplate="{DataTemplate local:ScanPage}" />
        </Tab>
       
    </TabBar>
    

 

    <FlyoutItem Title="Browse"
            Icon="tab_feed.png">
        <Tab>
            <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
        </Tab>
    </FlyoutItem>
    <FlyoutItem Title="About"
            Icon="tab_about.png">
        <Tab>
            <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
        </Tab>
    </FlyoutItem>

    

</Shell>

**Comment by Elevation72** [comment]
I see this issue in a new scaffolded project as well.

It's not an acceptable workaround, but if I add the following to my MainPage view in OnAppearing() the shell menu will open and when I manually close it the icon stays and the menu works after that.

Shell.Current.FlyoutIsPresented = true;

If I do not do this, I cannot access my flyout menu.

**Comment by Elevation72** [comment]
Replacing earlier comment with more complete info.  My experience is that I do not need to use tabs on the flyout menu to see this behavior.  This can be demonstrated on a new ShellDemo scaffolded project.

Repro steps:
1.  At app startup, the flyout menu default icon is not present
2.  Setting **FlyoutBehavior="Flyout"**  explicitly does not change the behavior
3.  Setting **FlyoutIcon="MyFlyoutIcon.png"** to an icon in my resource library does not change the behavior
4.  Setting **Shell.Current.FlyoutIsPresented = true** explicitly in code to force open the flyout menu, and then clicking on a menu item to return to the main page results in the menu now being present and working as expected.

It seems that somehow the Flyout Menu is in some sort of disabled or hidden state until this code is executed but I cannot find a property to enable it forcefully on startup.

**Comment by PureWeen** [comment]
Android or iOS?

@Elevation72 what version are you on? I tested your scenario with the latest 4.2 release and it all seems to work fine for me. 

We fixed some Flyout behavior in the latest 4.2 which most likely shifted the behavior back to being correct.

Looking at the original XAML on the issue the TabBar is the first element in the XAML. TabBar by default has the flyout disabled so it won't be visible.

If you want the Flyout to be visible change the *TabBar* to a FlyoutItem or specify ``` <TabBar Shell.FlyoutBehavior="Flyout" >```. The TabBar element is used if you want to have a page with no visible Flyout

I'm going to close this for now. If I get a repro seeing the issue on the latest version of forms then we'll reopen

**Comment by Elevation72** [comment]
I tested on both iOS and Android and seeing this in both places.  Your Shell.FlyoutBehavior attribute accomplished what I think may have previously default behavior (tab bar and flyout menu both enabled).  That resolves my issue...thanks.

**Comment by operationslazlo326** [comment]
We have the flyout items defined in the bottom part of the XAML. These are the menu items that should appear in the flyout menu. The Tab items defined at the top should appear at the bottom as tabs. The recommended fix Ido not believe applies to our objective. We wish to have tabs on the bottom and a flyout menu. This only appears to be possible on the nuget version I referenced above. Upgrading to anything new and not making  a single code change breaks our menu structure/behavior completely.

**Comment by PureWeen** [comment]
@operationslazlo326 

Change this to FlyoutItem

TabBar's don't persist between different flyout item navigations

If what I'm saying seems wrong can you please send a small gif of what you see working in one but not the other?

```XAML
<TabBar>
    <Tab Title="HOME" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:DealsPage}" />
    </Tab>
    <Tab Title="LOTTO" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:LotteryPage}" />
    </Tab>
    <Tab Title="GIFTCARD" Icon="{ext:Extensions.ImageResource com.enmarket.loyalty.consumer.Assets.Images.giftcard-icon.png}" StyleClass="TabText">
        <ShellContent ContentTemplate="{DataTemplate local:GiftCardPage}" />
    </Tab>
    <Tab Title="REWARDS" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
    </Tab>
    <Tab Title="SCAN" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:ScanPage}" />
    </Tab>
   
</TabBar>
```

I'm still not clear on what you're describing

**Comment by operationslazlo326** [comment]
Tabs at the bottom flyout menu to the left. What you just asked me to change to flyout items should effectively be the tabs located at the bottom of the shell. The original example I posted has both of them working in the older xamarin forms version.

**Comment by operationslazlo326** [comment]
Here is updated XAML, we did some refactoring to the flyout. But the bug is is still there. If we upgrade to any new version of xamarin forms we lose our flyout menu.


<?xml version="1.0" encoding="UTF-8"?>
<Shell xmlns="http://xamarin.com/schemas/2014/forms" 
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
       xmlns:d="http://xamarin.com/schemas/2014/forms/design"
       xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
       mc:Ignorable="d"
       xmlns:xyz="clr-namespace:Xamarin.Forms;assembly=Xamarin.Forms.Core"
       xmlns:my="clr-namespace:com.enmarket.loyalty.consumer"
       xmlns:local="clr-namespace:com.enmarket.loyalty.consumer.Views"
       Title="com.enmarket.loyalty.consumer"
       xmlns:ext="clr-namespace:com.enmarket.loyalty.consumer"
       x:Class="com.enmarket.loyalty.consumer.AppShell"
       >

    
    <Shell.Resources>
        <ResourceDictionary>
            <xyz:Color x:Key="NavigationPrimary">#70BB44</xyz:Color>
            <Style x:Key="BaseStyle" TargetType="Element">
                <Setter Property="Shell.BackgroundColor" Value="{StaticResource NavigationPrimary}" />
                <Setter Property="Shell.ForegroundColor" Value="White" />
                <Setter Property="Shell.TitleColor" Value="White" />
                <Setter Property="Shell.DisabledColor" Value="#B4FFFFFF" />
                <Setter Property="Shell.UnselectedColor" Value="#95FFFFFF" />
                <Setter Property="Shell.TabBarBackgroundColor" Value="{StaticResource NavigationPrimary}" />
                <Setter Property="Shell.TabBarForegroundColor" Value="White"/>
                <Setter Property="Shell.TabBarUnselectedColor" Value="#95FFFFFF"/>
                <Setter Property="Shell.TabBarTitleColor" Value="White"/>
                
            </Style>

            <Style TargetType="ShellItem" BasedOn="{StaticResource BaseStyle}" />
            <Style TargetType="TabBar" BasedOn="{StaticResource BaseStyle}" />

        </ResourceDictionary>
    </Shell.Resources>

    <Shell.FlyoutHeaderTemplate>
        <DataTemplate>
            <StackLayout Orientation="Vertical" VerticalOptions="Fill" HorizontalOptions="Start" BackgroundColor="White">
                <Image Source="{ext:Extensions.ImageResource com.enmarket.loyalty.consumer.Assets.Images.enlogo1.png}" />

                <my:FlyoutMenuItem Route="/home" Text="Home" Icon="tab_feed.png" />
                <my:FlyoutMenuItem Route="/wallet" Text="Wallet" Icon="tab_feed.png" />
                <my:FlyoutMenuItem Route="/about" Text="About" Icon="tab_feed.png" />

            </StackLayout>
        </DataTemplate>
    </Shell.FlyoutHeaderTemplate>

    
    
    <TabBar>
        <Tab Title="HOME" Icon="tab_feed.png" Route="home" FlyoutIcon="tab_feed.png">
            <ShellContent ContentTemplate="{DataTemplate local:DealsPage}" />
        </Tab>
        <Tab Title="LOTTO" Icon="{ext:Extensions.ImageResource com.enmarket.loyalty.consumer.Assets.Images.lotto-icon.png}" Route="lottery">
            <ShellContent ContentTemplate="{DataTemplate local:LotteryPage}" />
        </Tab>
        <Tab Title="GIFTCARD" Icon="{ext:Extensions.ImageResource com.enmarket.loyalty.consumer.Assets.Images.giftcard-icon.png}" StyleClass="TabText" Route="giftcard">
            <ShellContent ContentTemplate="{DataTemplate local:GiftCardPage}" />
        </Tab>
        <Tab Title="REWARDS" Icon="tab_feed.png" Route="rewards">
            <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
        </Tab>
        <Tab Title="SCAN" Icon="tab_feed.png" Route="scan">
            <ShellContent ContentTemplate="{DataTemplate local:DriverLicensePage}" />
        </Tab>
    </TabBar>
</Shell>









<?xml version="1.0" encoding="UTF-8"?>
<Grid xmlns="http://xamarin.com/schemas/2014/forms"
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
       x:Class="com.enmarket.loyalty.consumer.FlyoutMenuItem"
       x:Name="Root"
       HeightRequest="50">
    <Grid.GestureRecognizers>
        <TapGestureRecognizer Tapped="FlyoutItem_Tapped" />
    </Grid.GestureRecognizers>

    <Grid.ColumnDefinitions>
        <ColumnDefinition Width="Auto" />
        <ColumnDefinition Width="*" />
    </Grid.ColumnDefinitions>

    <Image Grid.Column="0" Source="{Binding Icon, FallbackValue=tab_feed.png}"
           Margin="20, 5, 20, 5"/>
    
    <Label Grid.Column="1" Text="{Binding Text, FallbackValue=Home test test}"
           VerticalOptions="Fill" FontSize="Medium"
           VerticalTextAlignment="Center" />
</Grid>

**Comment by PureWeen** [comment]
@operationslazlo326 when you are on a ShellItem defined as TabBar the flyout is hidden 

If you look at the source code for TabBar it's just a ShellItem with a disabled flyout
https://github.com/xamarin/Xamarin.Forms/blob/1dd94d6bae28c35b57acf49ef9b299e56351222f/Xamarin.Forms.Core/Shell/ShellItem.cs#L26

There was a bug with *4.2.0.709249* where disabling the flyout wasn't working but that was fixed in the most recent version

Can you past screen shots of what it looks like on *4.2.0.709249* when you are saying it's working how you want it to?

**Comment by operationslazlo326** [comment]
![image](https://user-images.githubusercontent.com/26207026/66067882-62649e00-e501-11e9-89d9-e74201e9ae1a.png)
![image](https://user-images.githubusercontent.com/26207026/66067900-6f818d00-e501-11e9-8ab2-6853c02669ef.png)

**Comment by operationslazlo326** [comment]
![image](https://user-images.githubusercontent.com/26207026/66068310-3695e800-e502-11e9-915a-f0a2860434ed.png)

**Comment by PureWeen** [comment]
Version 4.2.0.709249 broke FlyoutBehavior so that if it was disabled it still showed up 
https://github.com/xamarin/Xamarin.Forms/issues/7238

If you down grade to the version before *4.2.0.709249* you'll see the same behavior as the most recent behavior

TabBar sets the FlyoutBehavior to disabled
https://github.com/xamarin/Xamarin.Forms/issues/7656#issuecomment-537270298

If you want the flyout to be visible you need to set the FlyoutBehavior on the TabBar to somethinig other than Disabled or use a FlyoutItem instead of a TabBar

I'm also guessing you're running into an issue because right now you can't have a visible Flyout menu but have that page not visible in the flyout menu (You could use a template to hide it)

Once this issue is resolved (hopefully in 4.4/5) then you can show the flyout button
https://github.com/xamarin/Xamarin.Forms/issues/7230
